app_legacy/Site/Commands/UpdateUserTimestamps.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LegacyApp\Site\Commands;
+
+use Eloquent;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+use LegacyApp\Site\Models\User;
+
+class UpdateUserTimestamps extends Command
+{
+    protected $signature = 'ra-legacy:site:update-user-timestamps';
+
+    protected $description = 'Calculates missing and/or incorrect timestamps for users';
+
+    public function handle(): void
+    {
+        // Scott's account was created on the day of the domain registration
+        User::where('User', 'Scott')
+            ->update([
+                'Created' => '2012-10-03 13:37:00',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // https://web.archive.org/web/20121121150009/http://retroachievements.org/
+        User::whereIn('User', [
+            'Dave',
+            'Jimmy',
+            'Batman',
+            'Steve',
+            'Michael',
+            'poip',
+            'Sonic',
+            'qwer',
+            'Bob',
+            'Dan',
+            'Jim',
+            'qweqwe',
+            'qwe',
+        ])
+            ->update([
+                'Created' => '2012-11-21 13:37:00',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // Teario - first achievement unlock
+        User::where('User', 'Teario')
+            ->update([
+                'Created' => '2012-12-29 13:37:00',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // Same as nightgoat99
+        User::where('User', 'nightgoat')
+            ->update([
+                'Created' => '2013-03-02 04:17:05',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // Iterate through user accounts by ID, descending, fill gaps and correct order of timestamps.
+        Eloquent::unguard();
+        $maxId = User::whereNull('Created')->max('ID') + 1; // 106605
+        $users = User::withTrashed()
+            ->where('ID', '<=', $maxId)
+            ->orderByDesc('ID')
+            ->get(['ID', 'User', 'Created']);
+        $progressBar = $this->output->createProgressBar($users->count());
+        $progressBar->start();
+        $timestampMemento = null;
+        $updated = 0;
+        /** @var User $user */
+        foreach ($users as $user) {
+            // Apply previous timestamp if it's earlier than the current or no Created timestamp exists.
```

Comment:
Interesting idea. It pushed my creation timestamp back nearly an hour because the user created after me logged in quicker.

If you want to increase the accuracy a little more, you can cross-reference with the ForumTopicComment table. There's several accounts that have post history even earlier than their current Created timestamps.
```
select ua.ID, ua.User, ua.Created, min(ftc.DateCreated) from UserAccounts ua
left join ForumTopicComment ftc on ftc.Author=ua.User where ftc.DateCreated<ua.Created
group by 1,2,3 order by 1;
```
Note: TheJediSonic appears as an anomaly in those results. Presumably he was changed to be the owner of an official forum topic that greatly predates his account. You could fix the date on that entry, or just ignore him.

Can do the same with the Comment table (no oddities there):
```
select ua.ID, ua.User, ua.Created, min(c.Submitted) from UserAccounts ua
left join Comment c on c.UserID=ua.ID where c.Submitted<ua.Created
group by 1,2,3 order by 1;
```

Should do this first so the interpolation uses the updated data.

Reply:
Wonderful idea to cross-reference with forum topics and comments!

Made some adjustments in https://github.com/RetroAchievements/RAWeb/pull/1347/commits/de3536a44241efcad438c501442c541016b08f9d to take your feedback into account.

Fixed the ID to the first account that received a "real" creation timestamps which coincidentally excludes TheJediSonic's and other accounts whose diff to content creation would wildly exceed a reasonable threshold of a few hours/days.

Reply:
Seems there are some outliers that mess with the sequence. I'll add a threshold to make sure reassigned topics don't do that.